src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs

```diff
@@ -142,6 +149,8 @@ private void InternalDispose()
                 _status = Disposed;
             }
 
+            _timeoutOrCancellationSource?.Dispose();
```

Comment:
Dispose doesn't Cancel.  Do you want to also Cancel?  I'm just wondering about the comment on this method which suggests it does so.

Reply:
Interesting. I feel like this comment is just wrong. The way the method used to and still works is that if called while a ping request is in progress it sets `_disposeRequested` to true and then returns. Then when the ping request completes it calls `Finish` which completes the disposal.

It would be interesting if this comment implied that what `InternalDisposeCore` does (disposing the handles) can actually safely cancel outstanding Windows ping requests, but given that the [old .NET Framework version](https://referencesource.microsoft.com/#System/net/System/Net/NetworkInformation/ping.cs,131) seems to have the same behavior I'm not convinced. 

Do you agree? Should I just remove the comment? Should we change the code to start canceling the token once we've requested disposal to hurry outstanding requests along?

Reply:
> I feel like this comment is just wrong.

Looks like you're right. You can just delete it.